### PR TITLE
Allow GracefullyExit() to not presist on exiting

### PR DESCRIPTION
**Pull request by frenzibyte**
Useful when testing blocking exits.

### CODE REVIEW

File: osu.Game/OsuGame.cs [hunk 1]
```diff
@@ -934,14 +934,14 @@ protected override bool OnExiting()
 
         /// <summary>
         /// Use to programatically exit the game as if the user was triggering via alt-f4.
-        /// Will keep persisting until an exit occurs (exit may be blocked multiple times).
         /// </summary>
-        public void GracefullyExit()
+        /// <param name="repeatExit">Whether to keep presisting until an exit occurs (exit may be blocked multiple times).</param>
```
Comment (bdach) [inline osu.Game/OsuGame.cs:938]: Did you mean:

```suggestion
        /// <param name="repeatExit">Whether to keep persisting until an exit occurs (exit may be blocked multiple times).</param>
```

File: osu.Game/OsuGame.cs [hunk 2]
```diff
@@ -934,14 +934,14 @@ protected override bool OnExiting()
...
+        public void GracefullyExit(bool repeatExit = false)
```
Comment (bdach) [inline osu.Game/OsuGame.cs:939]: Why is `false` the default?

**Review by bdach — COMMENTED** [review]
I don't think I like this direction. To have this exposed just for testing feels too much. Why not just override `OsuGameTestScene.CreateGame()` and provide a subclass of `TestOsuGame` that does
```csharp
protected override bool OnExiting() => false;
```
so that the exit never actually happens? It seems like it'll be equivalent...?

**Comment by frenzibyte** [comment]
Closing as useless.

**Pull request closed without merging**